sql/core/src/main/scala/org/apache/spark/sql/jdbc/MsSqlServerDialect.scala

```diff
@@ -92,6 +92,7 @@ private case class MsSqlServerDialect() extends JdbcDialect {
               case o => inputToSQL(o)
             }
             visitBinaryComparison(e.name(), l, r)
+          case "CASE_WHEN" => super.visitCaseWhen(expressionsToStringArray(e.children())) + " = 1"
```

Comment:
let's override visitCaseWhen

Reply:
Problem with that is, we don't want to add this every time. We only do this when "CASE_WHEN" is Predicate. There are cases when "CASE_WHEN" is a ScalarExpression here, but not Predicate. So, adding this every time in visitCaseWhen would break every case when "CASE_WHEN" is not visited as a Predicate.

Reply:
Ah, I see. Thanks